rust/lance/src/dataset/write/merge_insert.rs

```diff
@@ -192,27 +192,13 @@ pub enum WhenMatched {
     DoNothing,
     /// The row is updated (similar to UpdateAll) only for rows where the expression evaluates to
     /// true
-    UpdateIf(Expr),
+    UpdateIf(String),
```

Comment:
Because each code path needs to parse the expression a little differently, I am deferring the expression parsing till later for now. When we delete the old code path later, we can remove this.